### ISSUE TITLE
Use company-tabnine with c-c++ layer

### ISSUE THREAD

**Issue by lipeng31** [syl20bnr/spacemacs#12920]
Hi. I'm new to emacs, and thanks a lot for the amazing work of spacemacs. Recently I got a problem when working with c-c++ layer and company-tabnine. The problem is that, when I use c-c++ layer, it seems to override the company-backends I set in my .spacemacs file, so I cannot use tabnine as the company backend. I did some web search and dived into c-c++ layer's code, but I can't figure out how to solve it.

I'm grateful for any suggestions.

**Comment by sei40kr** [comment]
I was working on TabNine backend
https://github.com/syl20bnr/spacemacs/pull/11985

`company-backends` is overridden here:
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bcompletion/auto-completion/funcs.el#L98

In addition, if you're using lsp, it be also overridden by here:
https://github.com/emacs-lsp/lsp-mode/blob/master/lsp-mode.el#L5335

**Comment by sei40kr** [comment]
The reason that I didn't merge the PR above is TabNine without semantic completion was terrible.
It seems TabNine has a feature to handle lsp inside, but it'll be difficult to use it in Spacemacs.

**Comment by lipeng31** [comment]
@sei40kr Thank you for the reply. After I filed this issue, I made a further try by slightly modifying two 'add-company-backends' lines in layers/+lang/c-c++/packages.el, adding company-tabnine to the argument. I think it's very dirty, but it works fine.

How ever this method only works for the c-c++ layer, I still can't use tabnine when writing other languages like python. I believe it relates to the completion layer as you mentioned, but that's really beyond my knowledge.

Looking forward to your ultimate solution. Thank you very much.
